File: RxDemo/src/main/java/com/vondear/rxdemo/activity/ActivityVibrate.java
package com.vondear.rxdemo.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.vondear.rxdemo.R;
import com.vondear.rxtools.RxVibrateTool;
import com.vondear.rxui.activity.ActivityBase;
import com.vondear.rxui.view.RxTitle;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * @author vondear
 */
public class ActivityVibrate extends ActivityBase {


    @BindView(R.id.btn_vibrate_once)
    Button mBtnVibrateOnce;
    @BindView(R.id.btn_vibrate_Complicated)
    Button mBtnVibrateComplicated;
    @BindView(R.id.btn_vibrate_stop)
    Button mBtnVibrateStop;
    @BindView(R.id.rx_title)
    RxTitle mRxTitle;

    private long[] temp = {100, 10, 100, 1000};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_vibrate);
        ButterKnife.bind(this);
        mRxTitle.setLeftFinish(mContext);
    }

    @OnClick({R.id.btn_vibrate_once, R.id.btn_vibrate_Complicated, R.id.btn_vibrate_stop})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btn_vibrate_once:
                RxVibrateTool.vibrateOnce(this, 2000);
                break;
            case R.id.btn_vibrate_Complicated:
                RxVibrateTool.vibrateComplicated(this, temp, 0);
                break;
            case R.id.btn_vibrate_stop:
                RxVibrateTool.vibrateStop();
                break;
        }
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityVibrate.kt
package com.tamsiree.rxdemo.activity

import android.os.Bundle
import com.tamsiree.rxdemo.R
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxkit.RxVibrateTool.vibrateComplicated
import com.tamsiree.rxkit.RxVibrateTool.vibrateOnce
import com.tamsiree.rxkit.RxVibrateTool.vibrateStop
import com.tamsiree.rxui.activity.ActivityBase
import kotlinx.android.synthetic.main.activity_vibrate.*

/**
 * @author tamsiree
 */
class ActivityVibrate : ActivityBase() {

    private val temp = longArrayOf(100, 10, 100, 1000)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_vibrate)
        setPortrait(this)
    }

    override fun initView() {
        rx_title.setLeftFinish(mContext)

        btn_vibrate_once.setOnClickListener { vibrateOnce(this, 2000) }
        btn_vibrate_Complicated.setOnClickListener { vibrateComplicated(this, temp, 0) }
        btn_vibrate_stop.setOnClickListener { vibrateStop() }
    }

    override fun initData() {

    }

}